pkg/registry/resource/devicetaintrule/strategy_test.go

```diff
@@ -47,40 +75,193 @@ func TestDeviceTaintRuleStrategy(t *testing.T) {
 
 func TestDeviceTaintRuleStrategyCreate(t *testing.T) {
 	ctx := genericapirequest.NewDefaultContext()
-	patch := patch.DeepCopy()
+	testcases := map[string]struct {
+		obj                   *resource.DeviceTaintRule
+		expectValidationError string
+		expectObj             *resource.DeviceTaintRule
+	}{
+		"simple": {
+			obj:       obj,
+			expectObj: obj,
+		},
+		"validation-error": {
+			obj: func() *resource.DeviceTaintRule {
+				obj := obj.DeepCopy()
+				obj.Name = "%#@$%$"
+				return obj
+			}(),
+			expectValidationError: metadataError,
+		},
+		"drop-status": {
+			obj:       objWithStatus,
+			expectObj: obj,
+		},
+		"set-generation": {
+			obj: func() *resource.DeviceTaintRule {
+				obj := obj.DeepCopy()
+				obj.Generation = 42 // Cannot be set by client on create, overwritten with 1.
+				return obj
+			}(),
+			expectObj: obj,
+		},
+	}
 
-	Strategy.PrepareForCreate(ctx, patch)
-	errs := Strategy.Validate(ctx, patch)
-	if len(errs) != 0 {
-		t.Errorf("unexpected error validating for create %v", errs)
+	for name, tc := range testcases {
+		t.Run(name, func(t *testing.T) {
+			obj := tc.obj.DeepCopy()
+			Strategy.PrepareForCreate(ctx, obj)
+			if errs := Strategy.Validate(ctx, obj); len(errs) != 0 {
+				assert.ErrorContains(t, errs[0], tc.expectValidationError, "the error message should have contained the expected error message")
```

Comment:
Does this have the same problems as we identified in https://github.com/kubernetes/kubernetes/pull/134962?

Reply:
I thought I had fixed it, but apparently not. Thanks for the reminder! Now fixed and pushed: https://github.com/kubernetes/kubernetes/compare/09869314e329a49ae8c168ef61c2e6cfcef62902..47607fb94c067f367926cf10a5d48fae0148e10d